modules/ROOT/pages/dataweave-cookbook-set-reader-writer-props.adoc

```diff
@@ -0,0 +1,192 @@
+
+The Read operation (`<file:read/>`) uses `outputMimeType='application/csv; separator=|; header=true'` to identify the input MIME type (`application/csv`), the CSV separator (`|`), and the header setting (`true`).
+
+The script in the Transform Message component (`<ee:transform-message/>`) returns the second row from the input CSV (`2345 , Achaval\,Mr. , arch`). The script uses `payload[1]` to select that row because DataWeave treats each row in the CSV as an index of an array. The output is comma-separated because the default CSV separator is a comma. The script omits the CSV header from the output by using the `output` directive (`output application/csv header = false`).
+
+The Logger component (`<logger/>`), which is set to `payload`, writes the comma-separated output as INFO in its `LoggerMessageProcessor` message: `2345 , Achaval\,Mr. , arch`. To avoid treating the comma before "Mr." as a separator, the CSV output escapes the comma with a backslash (`Achaval\,Mr.`).
+
+The following examples pass reader properties through the `outputMimeType` field.
+
+* xref:dataweave-formats-csv.adoc#example2[CSV example] that uses `streaming=true; header=true`
+* xref:dataweave-formats-json.adoc#example3[JSON example} that uses `streaming=true`
+
+[[ex04]]
+== Use a Mule Variable as a Configuration Value
+
+The following Mule flow transforms comma-separated CSV data into pipe-separated (`|`) CSV data. Instead of setting a literal value in the script, the example selects the pipe value from a xref:about-mule-variables.adoc[Mule variable], which is part of a Mule event that travels through data-processing components in the flow.
+
+The Scheduler component (`<scheduler/>`) in the example generates a Mule event each time it triggers the flow.
+
+The Set Variable component (`<set-variable/>`) creates a Mule variable named `delimiter` with the value `'|'`. In Studio, the value is created as an *fx* expression (`value="#['|']"`), rather than a simple string (`value="'|'"`).
+
+The Transform Message component (`<ee:transform-message/>`) contains a DataWeave script that reads comma-separated CSV input and uses the Mule variable to write pipe-separated CSV output.
+
+The Logger component (`<logger/>`), which is set to `payload`, writes the pipe-separated output as INFO in its `LoggerMessageProcessor` message: `macaroni | rigatoni | ravioli | spaghetti`.
+
+.Mule Flow:
+[source,xml,linenums]
+----
+<flow name="ex-use-mule-var-as-prop-config-val" >
+  <scheduler doc:name="Scheduler" >
+    <scheduling-strategy >
+      <fixed-frequency frequency="45" timeUnit="SECONDS"/>
+    </scheduling-strategy>
+  </scheduler>
+  <set-variable value="#['|']" doc:name="Set Variable" variableName="delimiter" />
+  <ee:transform doc:name="Transform Message" >
+    <ee:message>
+      <ee:set-payload><![CDATA[%dw 2.0
+var myVar = read("macaroni , rigatoni , ravioli , spaghetti", "application/csv" , {"header":false, separator:','})
+output application/csv with binary
```

Comment:
We should probably note that this is only available in 4.3.0+

Reply:
done